### ISSUE TITLE
AND policy

### ISSUE THREAD

**Issue by HuBaX** [open-telemetry/opentelemetry-collector-contrib#6802]
Currently, in the tailsamplingprocessor only one of the given policies has to be fulfilled for a trace to be sampled. With this approach, however, it is not possible to define complex rules for sampling a trace. It is not possible to combine multiple policies to build much more complex policies out of the existing policies. The composite policy only offers the possibility to link rate limiting with any other policy. 

I would like to have a policy that allows me to combine several of the existing policies so that the trace is not sampled until ALL of the specified policies are met. I imagine specifying this policy in the collector configuration file as follows: 
```
processors:
	tail_sampling: 
		... 
		policies: 
			[ 
		            {<other-policy>}, 
		            { 
		            name: and-policy, 
		            type: and, 
		            and: { 
			            combined_policy: 
				            [ 
					            {
                                                      name: probabilistic-policy,
                                                      type: probabilistic,
                                                      probabilistic: {sampling_percentage: 20}
                                                     },
                                                      { 
                                                        name: status-policy, 
                                                        type: status_code, 
                                                        status_code: {status_codes: [ERROR, UNSET]} 
                                                       } 
				            ] 
		            } 
			]
 ```
All policies listed under combined_policy must be met for the trace to be sampled. For example, policies could be set up where only 20% of the traces with an error status code are sampled.

I also think that this new policy will make the composite policy obsolete, since it is also possible to combine rate limiting with any other policy. 

I already have an implementation for the mentioned feature. However, before I create a PR for it, I need to test it extensively. 

I have already discovered a similar request in the following ticket: 
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/4760

**Comment by jpkrohling** [comment]
@pmm-sumo, was it you that had something similar ready as well?

**Comment by pmm-sumo** [comment]
@jpkrohling yes, we went with a slightly different way with [cascadingfilterprocessor](https://github.com/SumoLogic/opentelemetry-collector-contrib/tree/main/processor/cascadingfilterprocessor). Each policy (which we now call "trace accept rule") has a number of attributes which can be set and if present, are considered "AND". By having several policies, one could also achieve "OR" (either one policy matches or the other).

**Comment by mottibec** [comment]
I implemented this internally in our distro (i called it `combined` policy). If no one is working on this already, I'd love to upstream it to this repository.

**Comment by jpkrohling** [comment]
I think it would be a good idea to have a discussion during the next SIG call, to coordinate the work around this.

**Comment by jpkrohling** [comment]
From the SIG call: we are moving forward with this. In the future, we will consider using the libraries created as part of the new transform processor to provide a consistent way to express the filter.
